crates/rolldown/src/hmr/hmr_ast_finalizer.rs

```diff
@@ -0,0 +1,56 @@
+use oxc::{
+  allocator::Allocator,
+  ast::{VisitMut, ast, visit::walk_mut},
+};
+use rolldown_common::IndexModules;
+use rolldown_ecmascript_utils::AstSnippet;
+
+#[expect(unused)]
+pub struct HmrAstFinalizer<'me, 'ast> {
```

Comment:
Let me try to implement a minimum finalizer first, then let's see how we merge it to the exist finalizer.